e2e/cp_test.go

```diff
@@ -4596,3 +4596,160 @@ func TestCopyS3ObjectsWithIncludeExcludeFilter2(t *testing.T) {
 	expected := fs.Expected(t, expectedFileSystem...)
 	assert.Assert(t, fs.Equal(cmd.Dir, expected))
 }
+
+// cp --content-type "video/mp4" file s3://bucket/
+func TestCopySingleLocalFileToS3WithContentType(t *testing.T) {
+	t.Parallel()
+
+	s3client, s5cmd := setup(t)
+
+	bucket := s3BucketFromTestName(t)
+	createBucket(t, s3client, bucket)
+
+	const (
+		filename = "testfile.txt"
+		content  = "this is a test file"
+	)
+
+	workdir := fs.NewDir(t, t.Name(), fs.WithFile(filename, content))
+	defer workdir.Remove()
+
+	srcpath := workdir.Join(filename)
+	dstpath := fmt.Sprintf("s3://%v/", bucket)
+
+	cmd := s5cmd("cp", "--content-type", "video/mp4", srcpath, dstpath)
+
+	result := icmd.RunCmd(cmd)
+	result.Assert(t, icmd.Success)
+
+	// assert local filesystem
+	expected := fs.Expected(t, fs.WithFile(filename, content))
+	assert.Assert(t, fs.Equal(workdir.Path(), expected))
+
+	// assert s3
+	assert.Assert(t, ensureS3Object(s3client, bucket, filename, content, ensureContentType("video/mp4")))
+}
+
+// cp --content-type "video/avi" dir s3://bucket/
+func TestCopyMultipleFilesToS3BucketWithContentType(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("Files in Windows cannot contain glob(*) characters")
+	}
+	t.Parallel()
+
+	s3client, s5cmd := setup(t)
+
+	bucket := s3BucketFromTestName(t)
+	createBucket(t, s3client, bucket)
+
+	// create files with different content types
```

Comment:
no need for this comment, it's clear in the code.

Reply:
done.